airbyte-ci/connectors/pipelines/pipelines/helpers/changelog.py

```diff
@@ -0,0 +1,117 @@
+            + [line.to_str() for line in sorted_entries]
+            + self.original_markdown_lines[self.last_changelog_line_index + 1 :]
+        )
+        return "\n".join(new_lines) + "\n"
+
+    def __parse_markdown(self) -> None:
+        changelog_entry_re = (
+            "^\\| *(?P<version>[0-9]+\\.[0-9+]+\\.[0-9]+?) *\\| *"
+            + "(?P<day>[0-9]{4}-[0-9]{2}-[0-9]{2}) *\\| *"
+            + "\\[?(?P<pr_number1>[0-9]*)\\]? ?\\(https://github.com/airbytehq/airbyte/pull/(?P<pr_number2>[0-9]*)\\) *\\| *"
+            + "(?P<comment>[^ ].*[^ ]) *\\| *$"
+        )
+        changelog_header_line_index = -1
+        changelog_line_enumerator = enumerate(self.original_markdown_lines)
+        for line_index, line in changelog_line_enumerator:
+            if re.search(r"\| *Version *\| *Date *\| *Pull Request *\| *Subject *\|", line):
+                changelog_header_line_index = line_index
+                break
+        if changelog_header_line_index == -1:
+            raise Exception("Could not find the changelog section table in the documentation file.")
+        if self.original_markdown_lines[changelog_header_line_index - 1] != "":
+            raise Exception("Found changelog section table in the documentation file at line but there is not blank line before it.")
+        if not re.search(r"(\|-*){4}\|", next(changelog_line_enumerator)[1]):
+            raise Exception("The changelog table in the documentation file is missing the header delimiter.")
+        self.changelog_entries_start_line_index = changelog_header_line_index + 2
+
+        # parse next line to see if it needs to be cut
+        self.old_entries = set()
+        for line_index, line in changelog_line_enumerator:
+            changelog_entry_regexp = re.search(changelog_entry_re, line)
+            self.last_changelog_line_index = line_index
+            if not changelog_entry_regexp or changelog_entry_regexp.group("pr_number1") != changelog_entry_regexp.group("pr_number2"):
+                self.last_changelog_line_index = line_index - 1
+                break
+            entry_version = semver.VersionInfo.parse(changelog_entry_regexp.group("version"))
+            entry_date = datetime.datetime.strptime(changelog_entry_regexp.group("day"), "%Y-%m-%d").date()
+            entry_pr_number = int(changelog_entry_regexp.group("pr_number1"))
+            entry_comment = changelog_entry_regexp.group("comment")
+            changelog_entry = ChangelogEntry(entry_date, entry_version, entry_pr_number, entry_comment)
+            self.old_entries.add(changelog_entry)
+
+
+class ChangelogEntry:
+    def __init__(
+        self,
+        date: datetime.date,
+        version: semver.Version,
+        pr_number: int,
+        comment: str,
+    ) -> None:
+        self.date = date
+        self.version = version
+        self.pr_number = int(pr_number)
+        self.comment = comment
+
+    def to_str(self) -> str:
```

Comment:
```suggestion
    def to_markdown(self) -> str:
```
or 
```suggestion
    def __str__(self) -> str:
```
if you'd like to call `str(changelog_entry)`